crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -371,15 +372,17 @@ fn finalize_block<Block, Backend, Client>(
     });
 }
 
-/// Start an archiver that will listen for imported blocks and archive blocks at `K` depth,
+/// Crate an archiver task that will listen for importing blocks and archive blocks at `K` depth,
 /// producing pieces and segment headers (segment headers are then added back to the blockchain as
 /// `store_segment_header` extrinsic).
-pub fn start_subspace_archiver<Block, Backend, Client>(
+///
+/// NOTE: Archiver is doing blocking operations and must run in a dedicated task.
+pub fn create_subspace_archiver<Block, Backend, Client>(
     subspace_link: &SubspaceLink<Block>,
     client: Arc<Client>,
     telemetry: Option<TelemetryHandle>,
-    spawner: &impl sp_core::traits::SpawnEssentialNamed,
-) where
+) -> impl Future<Output = ()> + Send + 'static
```

Comment:
Why is this change needed?, please help understand

Reply:
It isn't see PR description